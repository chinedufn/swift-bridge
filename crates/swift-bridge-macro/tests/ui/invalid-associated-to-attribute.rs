//! # To Run
//! cargo test -p swift-bridge-macro -- ui trybuild=invalid-associated-to-attribute.rs

#[swift_bridge::bridge]
mod ffi {
    extern "Rust" {
        type SomeType;

        #[swift_bridge(associated_to = SomeType)]
        fn immutable_method(&self);

        #[swift_bridge(associated_to = SomeType)]
        fn mutable_method(&mut self);

        #[swift_bridge(associated_to = SomeType)]
        fn owned_method(self);
    }
}

pub struct SomeType;

impl SomeType {
    fn immutable_method(&self) {

    }
    fn mutable_method(&mut self) {

    }
    fn owned_method() {

    }
}

fn main() {}