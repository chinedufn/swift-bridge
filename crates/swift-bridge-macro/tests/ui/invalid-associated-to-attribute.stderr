error: The associated_to attribute can only be used on static methods.
  --> tests/ui/invalid-associated-to-attribute.rs:10:29
   |
10 |         fn immutable_method(&self);
   |                             ^^^^^

error: The associated_to attribute can only be used on static methods.
  --> tests/ui/invalid-associated-to-attribute.rs:13:27
   |
13 |         fn mutable_method(&mut self);
   |                           ^^^^^^^^^

error: The associated_to attribute can only be used on static methods.
  --> tests/ui/invalid-associated-to-attribute.rs:16:25
   |
16 |         fn owned_method(self);
   |                         ^^^^
